src/utils/cluster_utils.cpp

```diff
@@ -14,34 +15,32 @@ Eigen::VectorXd bayesmix::cluster_estimate(
   unsigned int n_data = alloc_chain.cols();
   Eigen::MatrixXd mean_diss = Eigen::MatrixXd::Zero(n_data, n_data);
   std::vector<Eigen::SparseMatrix<double> > all_diss;
+  progresscpp::ProgressBar bar(n_iter, 60);
+  std::cout << "(Computing mean dissimilarity... " << std::flush;
 
-  // Loop over iterations
-  for (size_t i = 0; i < n_iter; i++) {
-    // Find and all nonzero entries of the dissimilarity matrix
-    std::vector<Eigen::Triplet<double> > triplets_list;
-    triplets_list.reserve(n_data * n_data / 4);
-    for (size_t j = 0; j < n_data; i++) {
-      for (size_t k = 0; k < j; k++) {
-        if (alloc_chain(i, j) == alloc_chain(i, k)) {
-          triplets_list.push_back(Eigen::Triplet<double>(j, k, 1.0));
-        }
-      }
+  // Loop over pairs (i,j) of data points
+  for (int i = 0; i < n_data; i++) {
```

Comment:
Finalo comment here, I would move this code to another function 
```
Eigen::MatrixXd posterior_similarity_matrix(const Eigen::MatrixXd &alloc_chain)
```
it helps future developers